### PR TITLE
Added YARD docs in new mruby.h

### PR DESCRIPTION
**Pull request by Mav7**
Added documentation for mrb_funcall(), mrb_funcall_argv(), mrb_funcall_with_block(), and mrb_intern_cstr().

### CODE REVIEW

File: include/mruby.h [hunk 1]
```diff
@@ -263,28 +263,22 @@ MRB_API void mrb_define_method(mrb_state *mrb, struct RClass *cla, const char *n
  * Defines a class method.
  *
  * Example:
- *   # Ruby style
- *   class Foo
  *
- *     def Foo.bar
+ *     # Ruby style
+ *     class Foo
+ *      def Foo.bar
```
Comment (strcmp) [inline include/mruby.h]: It's one space indentation? I'm using phone. I think another space is needed for indentation.

**Comment by Mav7** [comment]
@strcmp I added one space indentation does it look right?

**Comment by strcmp** [comment]
Hi. Sorry. I wasn't clear but I'm on phone and can't explain properly now. It's worse than before. Please revert that commit.

**Comment by strcmp** [comment]
@Mav7 Line 267 is like this:
```ruby
# Ruby style
class Foo
 def Foo.bar
 end
end
```
There is one space used to indent but it should be two spaces. That's what I was trying to say. I don't think the new commit you added is good because it adds newlines and whitespace what seems in almost random way but it is my preference to not insert random newlines within a methods body to make code "look nicer".

**Comment by Mav7** [comment]
@strcmp I got rid of the new lines and white spaces from before. I'm sorry I thought that's what you meant at first. I think what added to the confusion is that for some weird reason when I pulled changes from this master branch into my own and tried generating docs; the formatting would apparently be off. So I had to make sure the generated YARD docs looked good. Let me know if it looks good.

**Comment by strcmp** [comment]
looks good, also fixes the original indentation issue. :+1:

**Comment by Mav7** [comment]
@strcmp Hey, would you like to help me document mruby's C API. I really appreciate your help on here and we could probably cover more ground together!

**Comment by strcmp** [comment]
hey again, yeah I would be happy to help. please drop me a line at strcmp [at] hmamail.com so we can discuss what you plan to document and how i could help. cheers.

**Comment by Mav7** [comment]
@strcmp It's hmamail.com? I just want to make sure I got that right.

**Comment by strcmp** [comment]
@Mav7 yeah that's right but i didn't give you right username. sorry. it is email service with very little hassle to sign up but i don't always use it, so i forgot. please email me at strcmp1@hmamail.com - look forward to it.

**Comment by Mav7** [comment]
@strcmp Ok just sent you an email. I can't wait to hear from you!

**Comment by strcmp** [comment]
@Mav7 thanks :) ill check the email soon. 
just a tip, i advise you to edit your comment and remove that email address.
robots can look for them, and send you nothing good(spam).

**Comment by Mav7** [comment]
@strcmp Wow thanks for letting me know.

**Comment by strcmp** [comment]
hi @Mav7 - sorry for the delay. I have replied by email.

**Comment by Mav7** [comment]
@strcmp Yeah it's fine. I just replied to the email.

**Pull request merged**